spark/src/main/scala/org/apache/spark/sql/comet/CometNativeWriteExec.scala

```diff
@@ -86,27 +154,149 @@ case class CometNativeWriteExec(nativeOp: Operator, child: SparkPlan, outputPath
     // Capture metadata before the transformation
     val numPartitions = childRDD.getNumPartitions
     val numOutputCols = child.output.length
+    val broadcastedCommitter = committer.map(c => sparkContext.broadcast(c))
```

Comment:
The committer should not use broadcast, because broadcast value is usually immutable.

Reply:
For example, `commiter.setupTask` will regenerate the internal commtier, so it should not be shared between tasks.

https://github.com/apache/spark/blob/161ed3d18dc346d3ad970b7a5997e42ea05b5206/core/src/main/scala/org/apache/spark/internal/io/HadoopMapReduceCommitProtocol.scala#L261

Reply:
Thanks for catching this. I removed the broadcast and captured the variable as part of the closure instead.